ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/Overview/usePlatformCveEntityCounts.ts

```diff
@@ -0,0 +1,27 @@
+import { gql, useQuery } from '@apollo/client';
+
+import { QuerySearchFilter } from '../../types';
+import { getRegexScopedQueryString } from '../../utils/searchUtils';
+
+const entityCountsQuery = gql`
+    query getPlatformCVEEntityCounts($query: String) {
+        platformCVECount(query: $query)
+        clusterCount(query: $query)
+    }
+`;
+
+export function usePlatformCveEntityCounts(querySearchFilter: QuerySearchFilter) {
```

Comment:
What do you think about making a more reusable hook for getting entity counts? We wouldn't need to duplicate the code with slight variations like in `useNodeCVEEntityCounts` and `usePlatformCVEEntityCounts`. We could have a hook called `useEntityCounts` which takes an array of entities we want counts for and the `querySearchFilter`. Then it returns the counts for those entities. I'd imagine it to look something like this:

```
const { data, loading, error } = useEntityCounts<{ platformCVECount: number, clusterCount: number }>(
    ['PlatformCVE', 'CLUSTER'],
    someQuerySearchFilter
);
```
Just brainstorming. We can look into the edge cases more too. This might be useful for Compliance too. It would be worth discussing with @bradr5 if the structure also aligns with what he expects for Compliance.

Reply:
I'm not sure that the additional abstraction for this use case would be worth it for entity counts. At least for Node/Platform CVEs, this hook would essentially be a mapping of string arguments to GQL fields right?

I think we might run into issues too with both:
- Mixing this abstraction with queries that pull data other than counts
- Implementing the GQL type generation I worked on a while back (if we decide to revisit this)

What I've been doing with GQL queries in VM 2.0 is trying to have fragments be colocated with the components that need them, and then use the components as the element of composition instead of the queries. For the counts, the query was simple enough and the component was close enough to the top of the tree that it felt like it made sense to just do this the simple way.